### ISSUE TITLE
Package search should search package `tags`, `description`, ... by default

### ISSUE THREAD

**Issue by jmid** [ocaml/ocaml.org#87]
First of all: thanks for your work on the new website! 😀 

I just gave the package search a spin. A bit of feedback: It seems the package search is only by name (or some substring).
I find that unfortunate from a UX point of view, as newcomers don't necessarily know the name of a relevant package beforehand (so they search).

Two examples:
- If I go to https://opam.ocaml.org/packages/ and search for `quickcheck` the result is 8 packages (all relevant).
  In comparison, https://v3.ocaml.org/packages/search?q=quickcheck only gives me 2 - which are both named something with `quickcheck`. 
- The story is the same for other search terms, e.g., `profiling`: https://opam.ocaml.org/packages/ shows me 6 packages
but https://v3.ocaml.org/packages/search?q=profiling shows 0 packages.

As such, newcomers may not find relevant packages or (even worse) think that there are few/no relevant OCaml packages available!
To improve the situation, I think searching additional opam package fields (`tags`, `description`, `synopsis`, ...)
should help. I would even argue that this should be the default.

Related search issues: #11 #38  #44

**Comment by jmid** [comment]
OK, I dug bit. The https://opam.ocaml.org/packages/ search seems to end here:
https://github.com/ocaml-opam/opam2web/blob/23366c29a505b96e9456cd4a1bf5027523a6a206/src/js_search/search.ml#L60
```ocaml
    if matches str name || matches str descr
    then show tr
    else hide tr
```
which searches the package description as well (client-side).

In comparison https://v3.ocaml.org/packages/search seems to search only package names (server-side):
https://github.com/ocaml/v3.ocaml.org-server/blob/7364ee120232e3ee7db8873fae232d95ae5a536a/lib/ocamlorg/package.ml#L379-L385

Modulo some forcing of the `Lazy.t` extraction of individual package info, I think something like the following could work:
```ocaml
  all_packages_latest ()
  |> List.filter (fun { name; info; _ } ->
         if contains (String.lowercase_ascii @@ Name.to_string name) pattern
            || contains (String.lowercase_ascii info.synopsis) pattern
            || contains (String.lowercase_ascii info.description) pattern       
            || List.exists (fun tag -> contains (String.lowercase_ascii tag) pattern) info.tags       
         then
           true
         else
           false)
```
Performance-wise this may be terrible if it happens to force too much. In that respect, I appreciate opam2web's design of filtering entirely client-side. As such, it may be easier to tackle in connection with #44.

**Comment by tmattio** [comment]
Closed in #131
